kernel/cClosure.ml

```diff
@@ -399,14 +399,13 @@ end = struct
         let cb = lookup_constant cst env in
         shortcut_irrelevant info (UVars.subst_instance_relevance u cb.const_relevance);
         let ts = RedFlags.red_transparent info.i_flags in
-        if TransparentState.is_transparent_constant ts cst then begin
-          match Cmap_env.find_opt cst env.symb_pats with
-          | Some r ->
-            let b = match [@ocaml.warning "-4"] cb.const_body with Symbol b -> b | _ -> assert false in
-            raise (NotEvaluableConst (HasRules (u, b, r)))
-          | None -> ();
+        if TransparentState.is_transparent_constant ts cst then match cb.const_body with
+        | Undef _ | Def _ | OpaqueDef _ | Primitive _ ->
           Def (constant_value_in u cb.const_body)
-        end else
+        | Symbol b ->
+          let r = Cmap_env.get cst env.symb_pats in
+          raise (NotEvaluableConst (HasRules (u, b, r)))
```

Comment:
Actually why do we do this here instead of in Table.constant_value_in (which currently has assert false for Symbol)?

Reply:
I think that this would have a different behaviour w.r.t. the transparent state.

Reply:
??

Reply:
Ah, I misread the code. Indeed that would be the same.

Reply:
On second thought it makes the constant_value_in more complex as it needs to take additional arguments (the env and constant name) that only matter for this case. I think we could clean this up but it requires more important API changes around the environment. So for the time being we should leave this as-is.